torch/signal/windows/windows.py

```diff
@@ -0,0 +1,126 @@
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('exponential_window', window_length, dtype, layout)
+
+    if window_length == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if window_length == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if periodic and center is not None:
+        raise ValueError('Center must be \'None\' for periodic equal True')
+
+    if center is None:
+        center = -(window_length if periodic else window_length - 1) / 2.0
+
+    k = torch.arange(start=center,
+                     end=center + window_length,
+                     dtype=dtype, layout=layout,
+                     device=device,
+                     requires_grad=requires_grad)
+
+    return torch.exp(-torch.abs(k) / tau)
+
+
+def cosine(window_length: int,
+           periodic: bool = True,
+           dtype: _dtype = None,
+           layout: _layout = torch.strided,
+           device: _device = None,
+           requires_grad: bool = False) -> Tensor:
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('cosine_window', window_length, dtype, layout)
+
+    if window_length == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if window_length == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    start = .5
+    k = torch.arange(start,
+                     start + window_length,
+                     dtype=dtype,
+                     layout=layout,
+                     device=device,
+                     requires_grad=requires_grad)
+    return torch.sin(torch.pi / (window_length + 1 if periodic else window_length) * k)
+
+
+def gaussian(window_length: int,
+             periodic: bool = True,
+             std: float = 0.5,
+             dtype: _dtype = None,
+             layout: _layout = torch.strided,
+             device: _device = None,
+             requires_grad: bool = False) -> Tensor:
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('cosine_window', window_length, dtype, layout)
+
+    if window_length == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if window_length == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    start = -(window_length if periodic else window_length - 1) / 2.0
+
+    k = torch.arange(start,
+                     start + window_length,
+                     dtype=dtype,
+                     layout=layout,
+                     device=device,
+                     requires_grad=requires_grad)
+
+    return torch.exp(-(k / std) ** 2 / 2)
```

Comment:
Actually we can do even better here as well. Note that `std` and the division by `2`  are also constants. We fuse that into the `arange` call using the `step=` parameter. This function should end with a `return torch.exp(-k**2)`.